src/dotnet/Program.cs

```diff
@@ -129,13 +139,7 @@ internal static int ProcessArgs(string[] args, ITelemetry telemetryClient)
                 exitCode = result.ExitCode;
             }
 
-            telemetryClient.TrackEvent(
-                command,
-                null,
-                new Dictionary<string, double>
-                {
-                    ["ExitCode"] = exitCode
```

Comment:
If we wait until the end to start sending telemetry, there will always be a 200-500 millisecond delay needed to complete the track event. If we wish to keep exitCode, we'd have to delay exiting the program until it was completed. 

There is no way around this, so we have to start sending earlier, which means we have to drop the exitCode.   Its not for a lack of wanting to know about exitCode, its about the constraint of minimizing impact on the runtime.